### PR TITLE
Assign each UIView element an identifier to make it visible under UI-Testing

### PR DESCRIPTION
**Pull request by Lucashuang0802**
## Pull request checklist

- [x] All tests pass. Demo project builds and runs.
- [x] I have resolved any merge conflicts.
- [x] I have followed the [coding style](https://github.com/jessesquires/HowToContribute#style-guidelines), and reviewed the [contributing guidelines](https://github.com/jessesquires/JSQMessagesViewController/blob/develop/.github/CONTRIBUTING.md). Confirmation: @Lucashuang0802

### CODE REVIEW

**Comment by jessesquires** [comment]
Thanks @Lucashuang0802 !

1. We don't need to set `accessibilityIdentifier`
2. `accessibilityLabel` should be a human-readable, localized string, not a class name.

File: JSQMessagesViewController/Views/JSQMessagesCollectionView.m [hunk 1]
```diff
@@ -47,6 +47,9 @@ @implementation JSQMessagesCollectionView
 - (void)jsq_configureCollectionView
 {
     [self setTranslatesAutoresizingMaskIntoConstraints:NO];
+    self.accessibilityIdentifier = NSStringFromClass([self class]);
+    self.accessibilityLabel = NSStringFromClass([self class]);
+    self.isAccessibilityElement = YES;
```
Comment (jessesquires) [inline JSQMessagesViewController/Views/JSQMessagesCollectionView.m]: all we need here is `self.isAccessibilityElement`

File: JSQMessagesViewController/Views/JSQMessagesCollectionViewCell.m [hunk 2]
```diff
@@ -111,6 +111,9 @@ - (void)awakeFromNib
     [super awakeFromNib];
 
     [self setTranslatesAutoresizingMaskIntoConstraints:NO];
+    self.accessibilityIdentifier = NSStringFromClass([self class]);
+    self.accessibilityLabel = NSStringFromClass([self class]);
+    self.isAccessibilityElement = YES;
```
Comment (jessesquires) [inline JSQMessagesViewController/Views/JSQMessagesCollectionViewCell.m]: all we need here is `self.isAccessibilityElement`

File: JSQMessagesViewController/Views/JSQMessagesTypingView.m [hunk 3]
```diff
@@ -54,6 +54,10 @@ - (void)setup
     _animationDuration = 1.33;
     _dotsColor = [UIColor lightGrayColor];
     _animateToColor = [UIColor grayColor];
+    
+    self.accessibilityIdentifier = NSStringFromClass([self class]);
+    self.accessibilityLabel = NSStringFromClass([self class]);
+    self.isAccessibilityElement = YES;
```
Comment (jessesquires) [inline JSQMessagesViewController/Views/JSQMessagesTypingView.m]: we can remove all of these

**Comment by jessesquires** [comment]
Most `accessibilityLabel`s should be set from the view controller.

For example:

- For a label, the `accessibilityLabel` is often the contents of the label.
- For a message cell, the `accessibilityLabel` can be something like "Message from <user>"

**Comment by Lucashuang0802** [comment]
I'm a little bit confused about the difference between these two properties. From the documentation, the property of `accessibilityIdentifier ` seems to be for UI testing: 
> An identifier can be used to uniquely identify an element in the scripts you write using the UI Automation interfaces. Using an identifier allows you to avoid inappropriately setting or accessing an element’s accessibility label.

And `accessibilityLabel` is for providing information to the assistive application. So, should I go with `accessibilityIdentifier` for UI testing?

**Comment by jessesquires** [comment]
Ah sorry, I misread this PR. 

Yes `accessibilityIdentifier` is for UI testing, but it should be unique. (For example, what you have here would set the same identifier for all cells.) 

[JSQDataSourcesKit](https://github.com/jessesquires/JSQDataSourcesKit) has a demo app with UI tests that you can look at for an example.

Let's update this PR with the following:

1. Let's **only** tackle cells right now. No other views.
2. Set `isAccessibilityElement` to `YES` in the super class
3. Set `accessibilityIdentifier` from `cellForItemAtIndexPath` to the index path values

**Comment by Lucashuang0802** [comment]
I added this line so it's easier to be query during UI recording like so:
```
XCUIElement *jsqmessagescollectionviewcellindexpathrow3Cell = app.collectionViews.cells[@"JSQMessagesCollectionViewCellIndexPathRow3"]
```
Otherwise, the UI recording will look like this which is not too obvious:
```
XCUIElement *timCookItIsUnitTestedFreeOpenSourceAndDocumentedCell = app.collectionViews.cells[@"Tim Cook: It is unit-tested, free, open-source, and documented."];
```

File: JSQMessagesViewController/Controllers/JSQMessagesViewController.m [hunk 4]
```diff
@@ -510,6 +510,7 @@ - (UICollectionViewCell *)collectionView:(JSQMessagesCollectionView *)collection
     }
 
     JSQMessagesCollectionViewCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:cellIdentifier forIndexPath:indexPath];
+    cell.accessibilityIdentifier = [NSString stringWithFormat:@"JSQMessagesCollectionViewCellIndexPathRow%ld", (unsigned long)indexPath.row];
```
Comment (jessesquires) [inline JSQMessagesViewController/Controllers/JSQMessagesViewController.m]: let's just make this `@"(%ld, %ld)", indexPath.section, indexPath.row`

that should be enough...

**Comment by Lucashuang0802** [comment]
It's updated! It's been a busy month 😃.

**Pull request merged**